### PR TITLE
Fixed #28 -- relaxed E301 for nested definitions

### LINKED ISSUE #28: Spurious E301 for nested function definitions
Simple nested functions without preceeding blank line are not a violation of PEP 8. It is up to the human to decide readability in this case.

pep8-0.6.1 will emit errors in cases where an extra blank line is not required or appropriate.

Example code, pep8_bug.py:

    """Demonstrate bug"""

    def safe_accessors(dictionary=False):
        """simplified from complex test code..."""
        if dictionary:
            _setattr, _getattr, _delattr = setitem, getitem, delitem
            def _hasattr(_dict, value):
                return value in _dict
        else:
            _setattr, _getattr, _delattr, _hasattr = (
                setattr, getattr, delattr, hasattr
            )
        return _setattr, _getattr, _delattr, _hasattr

pep8 output:

    pep8_bug.py:9:9: E301 expected 1 blank line, found 0

The correct output is to be silent (or at most emit a silence-able warning).

Reference: "PEP 8" / "Code lay-out" / "Blank Lines"

### PR DESCRIPTION
**Pull request by chrismedrela**

### CODE REVIEW

**Comment by memeplex** [comment]
Are you still interested in merging this one? In that case I will resolve the conficts.

File: pep8.py [hunk 1]
```diff
@@ -1319,6 +1323,14 @@ def check_logical(self):
         indent = first_line[:self.mapping[0][1][2][1]]
         self.previous_indent_level = self.indent_level
         self.indent_level = expand_indent(indent)
+
+        # Is this line inside a global (non-indented) function?
+        def is_not_global_class(line):
```
Comment (sigmavirus24) [inline pep8.py]: This isn't actually perfect. Take for example:

```py
# pkg/compat.py
"""Compat module for <library>."""
import collections

try:
    from collections import OrderedDict
except ImportError:
    class OrderedDict(collections.MutableMapping):
        # ...

try:
    from urlparse import urlparse
except ImportError:
    from urllib.parse import urlparse
```

```py
# pkg/something.py
from .compat import OrderedDict
#...
```

That's a global class and that line does not start with `class` instead it starts with `    class`.

**Comment by sigmavirus24** [comment]
FWIW, I agree with Florent that the checks shouldn't necessarily be relaxed but should instead be a separate error code. It gives users a lot more flexibility in how they configure pep8

**Comment by asottile** [comment]
#555 superseded this

**Pull request closed without merging**